test/fw/ptl/utils/pbs_logutils.py

```diff
@@ -286,20 +287,51 @@ def open_log(self, log, hostname=None, sudo=False):
         :type sudo: boolean
         :returns: A file instance
         """
+        readcmd = ['cat', log]
+        taillogs = 10000
+        if start:
+            i = 0
+            while(True):
+                i += 1
+                taillogs = 10000 * i
+                args = ['tail', '-n', str(taillogs), log]
+                args2 = ['head', '-n', '1']
+                process_tail = Popen(args, stdout=PIPE,
+                                     shell=False)
+                process_head = Popen(args2, stdin=process_tail.stdout,
+                                     stdout=PIPE, shell=False)
+                process_tail.stdout.close()
+                line = process_head.communicate()[0]
+                line = line.decode("utf-8")
+                ts = line.split(';')[0]
+                if '.' in ts:
+                    pattern = '%m/%d/%Y %H:%M:%S.%f'
+                else:
+                    pattern = '%m/%d/%Y %H:%M:%S'
+                epoch = int(time.mktime(time.strptime(ts, pattern)))
+                readcmd = ['tail', '-n', str(taillogs), log]
+                if start > epoch:
+                    break
+                elif taillogs > num_records:
+                    readcmd = ['cat', log]
+                    break
+
         try:
             if hostname is None or self.du.is_localhost(hostname):
                 if sudo:
-                    cmd = copy.copy(self.du.sudo_cmd) + ['cat', log]
+                    cmd = copy.copy(self.du.sudo_cmd) + readcmd
```

Comment:
No need for the copy.  When you add 2 lists together, you get a new list.  Now if you did
```
cmd = self.du.sudo_cmd
cmd += readcmd
```
You'd screw yourself because now sudo_cmd has changed as well.